ghostwriter/static/js/tinymce/plugins/ghostwriter-comments/plugin.min.js

```diff
@@ -0,0 +1,110 @@
+tinymce.PluginManager.add('ghostwriter-comments', function(editor) {
+    // Function to decode the users hex username into ascii
+    function hex2ascii(hexString) {
+        let decodedStr = '';
+        for (let i = 0; i < hexString.length; i += 2) {
+            const byte = parseInt(hexString.substr(i, 2), 16);
+            decodedStr += String.fromCharCode(byte);
+        }
+        return decodedStr
+    }
+
+    // Create Comment
+    function createComment(comment, selection) {
+        const commentNode = editor.selection.getNode();
+        let commentThread = readComments(commentNode);
+        commentThread.push({ author: username, comment: comment });
+        editor.execCommand('mceInsertContent', false, `<span class="comment" comment-data="${encodeURIComponent(JSON.stringify(commentThread))}">${selection}</span>`);
```

Comment:
This assumes the selection contains the entire comment. This is true when clicking on the comment, since the handler will select the area, but it isn't true for commenting on a section with a resolved comment, or when using the keyboard to manually select a section of text, and will lead to nested comment spans.